withastro/starlight
# Include built-in UI translation for regional docs

#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

This PR makes it possible for regional languages, e.g. `pt-BR` (Brazilian Portuguese) to use the built-in UI translation for the main language (e.g `pt`).

I've intentionally made that this would only work for the built-in translations, so you still can create an `i18n/pt-BR.json` file and override the built-in translation in case you want.

I've made a fairly simple case-insensitive function looking for the expected BCP-47 tag, we could throw an error if it isn't a valid string or make some more complex type checking for it, but I went simple to not overengineer for now.

TO-DO: docs! (let's see if we're going to change the API or something before)

#### Testing

The feature was tested with a copy branch of https://github.com/withastro/starlight/pull/469/, with the following steps:
- Add `getting-started.mdx` translation for `pt-BR`, the UI elements should be translated with the built-in UI i18n ✅ 
- Add `i18n/pt-BR.json`, the UI elements should ignore the built-in UI and use the user-provided file instead ✅
